src/sys/socket/mod.rs

```diff
@@ -217,7 +224,7 @@ pub enum SockProtocol {
     // The protocol number is fed into the socket syscall in network byte order.
     #[cfg(any(target_os = "android", target_os = "linux"))]
     #[cfg_attr(docsrs, doc(cfg(all())))]
-    EthAll = libc::ETH_P_ALL.to_be(),
+    EthAll = htons_i32(libc::ETH_P_ALL),
```

Comment:
Would this work instead?  It's a little bit simpler to inline it.
```suggestion
    EthAll = (libc::ETH_P_ALL as u16).to_be() as i32,
```

Reply:
I think it works the same, but it's a `const fn` so it's evaluated at compile time and won't require inlining, or did you just mean it's simpler to not have the function?

Happy to do it either way, but I *think* the other values need the same treatment on Linux, I just hadn't tested that far.

Reply:
Oh!  It would really help if you could test them too.  I suggested inlining that function just for readability's sake.